src/main/webapp/app/assessment-shared/assessment-warning/assessment-warning.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, SimpleChanges, Input, OnChanges } from '@angular/core';
+import * as moment from 'moment';
+import { Exercise } from 'app/entities/exercise';
+/*
+Warn instructors in the assessment and submission page
+if the due date is not over yet as they are allowed to assess submissions there
+ */
+@Component({
+    selector: 'jhi-assessment-warning',
+    templateUrl: './assessment-warning.component.html',
+})
+export class AssessmentWarningComponent implements OnChanges {
+    @Input()
+    exercise: Exercise;
+    currentDate: moment.MomentInput;
+    isBeforeDueDate = false;
+
+    ngOnChanges(changes: SimpleChanges): void {
+        const dueDate = this.exercise.dueDate;
```

Comment:
So I guess this should be only triggered when `exercise` changes. Can we move it to the `exercise` setter?